core/field_variable.js

```diff
@@ -217,31 +259,24 @@ Blockly.FieldVariable.dropdownCreate = function() {
  */
 Blockly.FieldVariable.prototype.onItemSelected = function(menu, menuItem) {
   var id = menuItem.getValue();
-  // TODO(marisaleung): change setValue() to take in an ID as the parameter.
-  // Then remove itemText.
-  var itemText;
   if (this.sourceBlock_ && this.sourceBlock_.workspace) {
     var workspace = this.sourceBlock_.workspace;
-    var variable = workspace.getVariableById(id);
-    // If the item selected is a variable, set itemText to the variable name.
-    if (variable) {
-      itemText = variable.name;
-    } else if (id == Blockly.RENAME_VARIABLE_ID) {
+    if (id == Blockly.RENAME_VARIABLE_ID) {
       // Rename variable.
-      var currentName = this.getText();
-      variable = workspace.getVariable(currentName);
-      Blockly.Variables.renameVariable(workspace, variable);
+      Blockly.Variables.renameVariable(workspace, this.variable_);
       return;
     } else if (id == Blockly.DELETE_VARIABLE_ID) {
       // Delete variable.
-      workspace.deleteVariable(this.getText());
+      workspace.deleteVariableById(this.variable_.getId());
       return;
     }
 
-    // Call any validation function, and allow it to override.
-    itemText = this.callValidator(itemText);
-  }
-  if (itemText !== null) {
-    this.setValue(itemText);
+    // TODO: Call any validation function, and allow it to override.
+    // For now it's unclear whether the validator should act on the id.
+    //var validatedId = this.callValidator(variable.getId());
   }
+  // if (variable.getId() !== null) {
```

Comment:
Can this commented out code be removed?

Reply:
I'm hoping to resolve it before I merge the rest of this into develop.  If not I'll make an issue for it and remove this code for that merge.